src/Lean/Meta/IndPredBelow.lean

```diff
@@ -374,6 +378,183 @@ where
       forallBoundedTelescope rest (some 1) fun ys rest =>
       loop xs rest belowIndices xIdx (yIdx + 1)
 
+private def belowType (motive : Expr) (xs : Array Expr) (idx : Nat) : MetaM $ Name × Expr := do
+  (←inferType xs[idx]).withApp fun type args => do
+  let indName := type.constName!
+  let indInfo ← getConstInfoInduct indName
+  let belowArgs := args[:indInfo.numParams] ++ #[motive] ++ args[indInfo.numParams:] ++ #[xs[idx]]
+  let belowType := mkAppN (mkConst (indName ++ `below) type.constLevels!) belowArgs
+  (indName, belowType)
+
+partial def mkBelowMatcher
```

Comment:
Could you please add a comment explaining this function (arguments, the resulting pair, etc)?